errors.go

```diff
@@ -92,11 +92,16 @@ func checkForErrors(res *http.Response) error {
 		res.Body.Close()
 	}()
 
+	out, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return unknownServerError(res.StatusCode, err.Error())
+	}
+
 	switch res.StatusCode {
 	case 400:
 		bodyMap := make(map[string]interface{})
-		if err := json.NewDecoder(res.Body).Decode(&bodyMap); err != nil {
-			return unknownError(res.StatusCode, err)
+		if err := json.Unmarshal(out, &bodyMap); err != nil {
+			return unknownServerError(res.StatusCode, err.Error())
```

Comment:
ok, that's fair. I was just saying that it may return _only_ the json decoding error and not the full response body. maybe it could be something the below?

```go
fmt.Sprintf("error decoding json response (%s): %s", err, string(out))
```